driver/custom_endpoint_monitor.cc

```diff
@@ -47,55 +46,81 @@ AWS_SDK_HELPER SDK_HELPER;
 
 CACHE_MAP<std::string, std::shared_ptr<CUSTOM_ENDPOINT_INFO>> CUSTOM_ENDPOINT_MONITOR::custom_endpoint_cache;
 
-CUSTOM_ENDPOINT_MONITOR::CUSTOM_ENDPOINT_MONITOR(const std::shared_ptr<HOST_INFO>& custom_endpoint_host_info,
+CUSTOM_ENDPOINT_MONITOR::CUSTOM_ENDPOINT_MONITOR(const std::shared_ptr<TOPOLOGY_SERVICE> topology_service,
+                                                 const std::string& custom_endpoint_host,
                                                  const std::string& endpoint_identifier, const std::string& region,
-                                                 DataSource* ds, bool enable_logging)
-  : custom_endpoint_host_info(custom_endpoint_host_info),
-    endpoint_identifier(endpoint_identifier),
-    region(region),
-    enable_logging(enable_logging) {
+                                                 long long refresh_rate_nanos, ctpl::thread_pool& thread_pool,
+                                                 bool enable_logging)
+    : topology_service(topology_service),
+      custom_endpoint_host(custom_endpoint_host),
+      endpoint_identifier(endpoint_identifier),
+      region(region),
+      refresh_rate_nanos(refresh_rate_nanos),
+      thread_pool(thread_pool),
+      enable_logging(enable_logging) {
   if (enable_logging) {
     this->logger = init_log_file();
   }
 
-  ++SDK_HELPER;
+  this->run();
+}
 
-  Aws::RDS::RDSClientConfiguration client_config;
-  if (!region.empty()) {
-    client_config.region = region;
+#ifdef UNIT_TEST_BUILD
+CUSTOM_ENDPOINT_MONITOR::CUSTOM_ENDPOINT_MONITOR(const std::shared_ptr<TOPOLOGY_SERVICE> topology_service,
+                                                 const std::string& custom_endpoint_host,
+                                                 const std::string& endpoint_identifier, const std::string& region,
+                                                 long long refresh_rate_nanos, ctpl::thread_pool& thread_pool,
+                                                 bool enable_logging, std::shared_ptr<Aws::RDS::RDSClient> client)
```

Comment:
I see that you added the `client` parameter to this version of the constructor but I don't see it being used in any of the initializers below. How is this used?

Reply:
This is used in the unit test